File: read_glb.js
const fs = require('fs');
try {
    const buf = fs.readFileSync('models/iphone.glb');
    const magic = buf.toString('utf8', 0, 4);
    if (magic !== 'glTF') throw new Error('Not GLB');
    const chunkLen = buf.readUInt32LE(12);
    const chunkType = buf.toString('utf8', 16, 20);
    if (chunkType !== 'JSON') throw new Error('Not JSON');
    const jsonStr = buf.toString('utf8', 20, 20 + chunkLen);
    const gltf = JSON.parse(jsonStr);
    
    console.log("--- Nodes ---");
    gltf.nodes.forEach((n, i) => console.log(`Node ${i}: ${n.name}`));
    
    console.log("\n--- Materials ---");
    gltf.materials.forEach((m, i) => console.log(`Material ${i}: ${m.name}`));
} catch (e) {
    console.error(e);
}

File: read_glb.py
import struct, json
with open('models/iphone.glb', 'rb') as f:
    magic, version, length = struct.unpack('<4sII', f.read(12))
    if magic != b'glTF': print('Not a GLB')
    chunk_len, chunk_type = struct.unpack('<II', f.read(8))
    if chunk_type != b'JSON': print('First chunk not JSON')
    json_bytes = f.read(chunk_len)
    gltf = json.loads(json_bytes)
    
    # Print node names and mesh indices
    for i, node in enumerate(gltf.get('nodes', [])):
        print(f"Node {i}: name='{node.get('name', 'N/A')}', mesh={node.get('mesh', 'N/A')}")
    
    for i, mat in enumerate(gltf.get('materials', [])):
        print(f"Material {i}: name='{mat.get('name', 'N/A')}'")
